Source/Diagnostics/ComputeDiagFunctors/BackTransformParticleFunctor.cpp

```diff
@@ -15,29 +15,37 @@
 #include <AMReX_Print.H>
 #include <AMReX_BaseFwd.H>
 
-SelectParticles::SelectParticles (const WarpXParIter& a_pti, TmpParticles& tmp_particle_data,
-                                  amrex::Real current_z_boost, amrex::Real old_z_boost,
-                                  int a_offset)
+SelectParticles::SelectParticles (
+    const WarpXParticleContainer& pc,
+    WarpXParIter& a_pti,
+    amrex::Real current_z_boost,
+    amrex::Real old_z_boost,
+    int a_offset
+)
```

Comment:
@atmyers 

Before I proceed in a similar way to remove `tmp_particle_data` from `CopyParticleAttribs`, do you agree with how I did it here? 

I am now passing the `WarpXParticleContainer` as an argument to this function in order to be able to call `getParticleComps` from it and use the result when calling `GetAttribs` from the particle iterator.

Does that make sense or do you see another/better way to do this?